packages/commonwealth/client/scripts/views/components/component_kit/cw_vote_button.tsx

```diff
@@ -0,0 +1,94 @@
+/* @jsx m */
+
+import m from 'mithril';
+
+import 'components/component_kit/cw_vote_button.scss';
+
+import { formatNumberShort } from 'adapters/currency';
+import { CWText } from './cw_text';
+import { ComponentType } from './types';
+import { CWIcon } from './cw_icons/cw_icon';
+import { getClasses } from './helpers';
+
+type VoteButtonAttrs = {
+  updateVoteCount: (newCount: number) => void;
+  voteCount: number;
+};
+
+export class CWVoteButton implements m.ClassComponent<VoteButtonAttrs> {
+  private isHoveringUpvote: boolean;
+  private isHoveringDownvote: boolean;
+  private initialVoteCount: number;
+
+  oninit(vnode) {
+    this.initialVoteCount = vnode.attrs.voteCount;
+  }
+
+  view(vnode) {
+    const { updateVoteCount, voteCount } = vnode.attrs;
+
+    const handleVoteChange = (newCount: number) => {
+      updateVoteCount(newCount);
+    };
+
+    return (
+      <div
+        class={getClasses<{
+          isHoveringUpvote: boolean;
+          isHoveringDownvote: boolean;
+          hasUpvoted: boolean;
+          hasDownvoted: boolean;
+        }>(
+          {
+            isHoveringUpvote: this.isHoveringUpvote,
+            isHoveringDownvote: this.isHoveringDownvote,
+            hasUpvoted: voteCount === this.initialVoteCount + 1,
+            hasDownvoted: voteCount === this.initialVoteCount - 1,
+          },
+          ComponentType.VoteButton
+        )}
+      >
+        <CWIcon
+          iconName="upvote"
+          iconSize="small"
+          onclick={() => {
+            voteCount === this.initialVoteCount + 1
```

Comment:
should add a TODO comment in here about creating the appropriate reaction in the db